### PR TITLE
23967 Removed spinner for approved continuation in drafts

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#23967

*Description of changes:*
- Removed spinner for approved continuation in drafts

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by JazzarKarim** [comment]
Watch out for lint errors Leo!

**Review by JazzarKarim — APPROVED** [review]

File: src/components/Dashboard/TodoList.vue [hunk 1]
```diff
@@ -343,7 +343,11 @@
                 <!-- NB: blocks below are mutually exclusive, and order is important -->
 
                 <!-- this loading button pre-empts all buttons below -->
-                <template v-if="inProcessFiling === item.filingId">
+                <!-- special case to not show spinner for approved continuation in drafts -->
+                <template 
+                  v-if="inProcessFiling === item.filingId && 
+                    !(EnumUtilities.isTypeContinuationIn(item) && EnumUtilities.isStatusApproved(item))"
+                  >
```
Comment (severinbeauvais) [inline src/components/Dashboard/TodoList.vue:350]: Why not change the logic for `inProcessFiling` in the code instead?

I think that, with this change, the 5x polling is still being done in the background. It doesn't need to be done for a continuation in authorization.

Comment (leodube-aot) [reply src/components/Dashboard/TodoList.vue]: Good point, I've moved the check into the `highlighTask` method since the `inProcessFiling` is just a simple number type. 

I also noticed the bug was occurring for continuation in filings with status `ACTIVE`. I'm not sure what other status types we can expect for continuation in filings, so I changed it to only display the spinner for `PENDING` status continuation ins. Let me know if that makes sense to do or not.

Comment (severinbeauvais) [reply src/components/Dashboard/TodoList.vue]: Thanks for the change. Looks good to me.

**Review by severinbeauvais — APPROVED** [review]

File: src/components/Dashboard/TodoList.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/Dashboard/TodoList.vue]: > I also noticed the bug was occurring for continuation in filings with status `ACTIVE`. I'm not sure what other status types we can expect for continuation in filings, so I changed it to only display the spinner for `PENDING` status continuation ins. Let me know if that makes sense to do or not.

The continuation in draft had a status of Active? That's unexpected. I think the normal flow is:
DRAFT → AWAITING_REVIEW → APPROVED → PENDING → PAID → COMPLETED

Basically, we only want the retry to work for the PENDING (ie, while we wait for PAID).

Ask if this doesn't make sense.

Comment (leodube-aot) [reply src/components/Dashboard/TodoList.vue]: Right my bad, I was looking at the wrong status field. I was looking at the one in `continationIn` rather than the `header`.

Screenshot 2024-11-06 at 3 40 54 PM

That makes sense though, thanks for the clarification!

**Comment by eason-pan-bc** [comment]
maybe consider adjusting the version number and run `npm i` again? I just merged a PR in.

**Review by eason-pan-bc — APPROVED** [review]
